sysl2/sysl/ints_view.go

```diff
@@ -0,0 +1,587 @@
+	nameMap := map[string]string{}
+	clusters := map[string][]string{}
+	for _, v := range apps {
+		cluster := strings.Split(v, " :: ")
+		if len(cluster) > 1 {
+			clusters[cluster[0]] = append(clusters[cluster[0]], v)
+		}
+	}
+
+	for k, v := range clusters {
+		if len(v) <= 1 {
+			delete(clusters, k)
+		}
+		for _, s := range v {
+			nameMap[s] = strings.Split(s, " :: ")[1]
+		}
+	}
+
+	for k, apps := range clusters {
+		fmt.Fprintf(v.stringBuilder, "package \"%s\" {", k)
+		fmt.Fprintln(v.stringBuilder)
+		for _, n := range apps {
+			v.VarManagerForComponent(n, nameMap)
+		}
+		fmt.Fprintln(v.stringBuilder, "}")
+	}
+
+	return nameMap
+}
+
+func (v *IntsDiagramVisitor) generateStateView(args *Args, viewParams viewParams, params *IntsParam) string {
+
+	fmt.Fprintln(v.stringBuilder, "@startuml")
+	if viewParams.diagramTitle != "" {
+		fmt.Fprintln(v.stringBuilder, "title "+viewParams.diagramTitle)
+	}
+	fmt.Fprintln(v.stringBuilder, "left to right direction")
+	fmt.Fprintln(v.stringBuilder, "scale max 16384 height")
+	fmt.Fprintln(v.stringBuilder, "hide empty description")
+	fmt.Fprintln(v.stringBuilder, "skinparam state {")
+	fmt.Fprintln(v.stringBuilder, "  BackgroundColor FloralWhite")
+	fmt.Fprintln(v.stringBuilder, "  BorderColor Black")
+	fmt.Fprintln(v.stringBuilder, "  ArrowColor Crimson")
+	if viewParams.highLightColor != "" {
+		fmt.Fprintln(v.stringBuilder, "  BackgroundColor<<highlight>> "+viewParams.highLightColor)
+	}
+	if viewParams.arrowColor != "" {
+		fmt.Fprintln(v.stringBuilder, "  ArrowColor "+viewParams.arrowColor)
+	}
+
+	if viewParams.indirectArrowColor != "" && viewParams.indirectArrowColor != "none" {
+		fmt.Fprintln(v.stringBuilder, "  ArrowColor<<indirect>> "+viewParams.indirectArrowColor)
+		fmt.Fprintln(v.stringBuilder, "  ArrowColor<<internal>> "+viewParams.indirectArrowColor)
+	}
+	fmt.Fprintln(v.stringBuilder, "}")
+
+	v.buildClusterForStateView(params.integrations, viewParams.restrictBy)
+	var processed []string
+	for _, dep := range params.integrations {
+		appA := dep.Self.Name
+		appB := dep.Target.Name
+		epA := dep.Self.Endpoint
+		epB := dep.Target.Endpoint
+		_, restrictByAppA := v.mod.Apps[appA].Attrs[viewParams.restrictBy]
+		_, restrictByAppB := v.mod.Apps[appB].Attrs[viewParams.restrictBy]
+		_, restrictByEpA := v.mod.Apps[appA].Endpoints[epA].Attrs[viewParams.restrictBy]
+		_, restrictByEpB := v.mod.Apps[appB].Endpoints[epB].Attrs[viewParams.restrictBy]
+		if viewParams.restrictBy != "" && !restrictByAppA && !restrictByAppB {
+			continue
+		}
+		if viewParams.restrictBy != "" && !restrictByEpA && !restrictByEpB {
+			continue
+		}
+		matchApp := appB
+		matchEp := epB
+		label := ""
+		needsInt := appA != matchApp
+
+		var pubSubSrcPtrns []string
+		if v.mod.Apps[appA].Endpoints[epA].Attrs["patterns"] != nil {
+			for _, v := range v.mod.Apps[appA].Endpoints[epA].Attrs["patterns"].GetA().Elt {
+				pubSubSrcPtrns = append(pubSubSrcPtrns, v.GetS())
+			}
+		}
+
+		tgtPtrns := MakeStrSet()
```

Comment:
please use better names, is this `targetPatterns`?

Reply:
changed to targetPatterns